### DISCUSSION TITLE
Extending simple boolean for counts

### DISCUSSION THREAD

**Discussion by JHBalaji** [pyparsing/pyparsing#469] [Q&A]
Hello there
I am trying to build a parser following this [example](https://gist.github.com/kampta/6f309e03c7ebf72838eca330adaa2fe2). But how would I go about implementing say count of Booleans in my matches?

```
p, q, r, k = True, False, True, True

For example,
 ("p or q", 2) is False - resolves to [T, F]
("p or q or r", 2) is True - resolves to [T, F, T]
("(p or q or r) or (p or q)", 1) is False - as (p or q or r) or (p or q) resolves to (True, True) equating to 2. 
 ("(p or k) or q", 1) is True - resolves as [T, F]
```

As you can see, I am interested in counting booleans in the last level while resolving.

**Comment by ptmcg** [comment]
All your examples are "or" operations. Does this mean you interpret "p and q or r" as [F, T]?

**Reply by JHBalaji** [reply]
Yes. I follow BODMAS, the precedence of bracket and multiplication and addition here.

So, it will be resolved like, (p and q) or r which results in [F, T]. The second value in tuple would be 1, if I'm looking for exactly one boolean to be True.

**Comment by ptmcg** [comment]
Also, what is the significance of the second element (the integer) in each of your example tuples?

**Reply by JHBalaji** [reply]
The second element is the count of TRUE booleans, I am expecting in the expression. For example, ("p or q or r", 2) is True, has two True's and one False, so it evaluates to True.
